### PR TITLE
[counterpoll] add port buffer drop group

### PR DESCRIPTION
**Pull request by mykolaf**
Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

Introducing separate FC group for port lever buffer drop counters as discussed in Azure/sonic-swss#1308
**- What I did**

**- How I did it**

**- How to verify it**

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

### CODE REVIEW

File: counterpoll/main.py [hunk 1]
```diff
@@ -81,6 +83,46 @@ def disable():
...
+    """ Port buffer drop  counter commands """
+
+@port_buffer_drop.command()
+@click.argument('poll_interval', type=click.IntRange(30000, 300000))
+def interval(poll_interval):
+    """
+    Set port_buffer_drop counter query interval
+    This counter group causes high CPU usage when polled,
+    hence the allowed interval is between 30s and 300s.
+    This is a ahort term solution and
```
Comment (daall) [inline counterpoll/main.py:98]:
```suggestion
    This is a short term solution and
```
typo

File: counterpoll/main.py [hunk 2]
```diff
@@ -81,6 +83,46 @@ def disable():
...
+    should be changed once the performance is enhanced
+    """
+    configdb = swsssdk.ConfigDBConnector()
+    configdb.connect()
+    port_info = {}
+    if poll_interval is not None:
```
Comment (daall) [inline counterpoll/main.py:104]:
```suggestion
    if poll_interval:
```
Since `poll_interval` can never == 0 can we do it this way?

**Comment by lguohan** [comment]
need unit test

File: tests/counterpoll_test.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        print(result.output)
+        assert result.output == expected_counterpoll_show
+
+    def test_port_buffer_drop_interval(self):
+        runner = CliRunner()
+        result = runner.invoke(counterpoll.cli.commands["port-buffer-drop"].commands["interval"], ["30000"])
+        print(result.output)
+        assert result.exit_code == 0
+
+    def test_port_buffer_drop_interval_to_short(self):
```
Comment (daall) [inline tests/counterpoll_test.py:46]:
```suggestion
    def test_port_buffer_drop_interval_too_short(self):
```

File: counterpoll/main.py [hunk 4]
```diff
@@ -81,6 +83,46 @@ def disable():
...
+    port_info['FLEX_COUNTER_STATUS'] = 'enable'
+    configdb.mod_entry("FLEX_COUNTER_TABLE", PORT_BUFFER_DROP, port_info)
+
+@port_buffer_drop.command()
+def disable():
+    """ Disable port counter query """
+    configdb = swsssdk.ConfigDBConnector()
+    configdb.connect()
+    port_info = {}
+    port_info['FLEX_COUNTER_STATUS'] = 'disable'
```
Comment (daall) [inline counterpoll/main.py:123]: Can we use the DISABLE and ENABLE macros above?

File: tests/counterpoll_test.py (hunk 3 above)
Comment (volodymyrsamotiy) [reply tests/counterpoll_test.py]: Fixed

File: counterpoll/main.py (hunk 4 above)
Comment (volodymyrsamotiy) [reply counterpoll/main.py]: Fixed

**Comment by daall** [comment]
retest this please

**Pull request merged**